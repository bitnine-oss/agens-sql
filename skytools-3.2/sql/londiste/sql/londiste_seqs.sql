
set client_min_messages = 'warning';
\set VERBOSITY 'terse'

--
-- sequences
--

create sequence masterseq;
create sequence slaveseq;


select * from pgq_node.register_location('seqroot', 'rnode', 'dbname=db', false);
select * from pgq_node.create_node('seqroot', 'root', 'rnode', 'londiste_root', null::text, null::int8, null::text);

select * from londiste.local_add_seq('seqroot', 'masterseq');
select * from londiste.local_add_seq('seqroot', 'masterseq');
select * from londiste.root_check_seqs('seqroot');
select * from londiste.local_remove_seq('seqroot', 'masterseq');
select * from londiste.local_remove_seq('seqroot', 'masterseq');

select * from londiste.get_seq_list('seqroot');

select ev_id, ev_type, ev_data, ev_extra1 from pgq.event_template where ev_type like '%seq%';

-- subscriber
select * from pgq_node.register_location('seqbranch', 'subnode', 'dbname=db', false);
select * from pgq_node.register_location('seqbranch', 'rootnode', 'dbname=db', false);
select * from pgq_node.create_node('seqbranch', 'branch', 'subnode', 'londiste_branch', 'rootnode', 1, null::text);

select * from londiste.local_add_seq('seqbranch', 'masterseq');
select * from londiste.global_update_seq('seqbranch', 'masterseq', 5);
select * from londiste.local_add_seq('seqbranch', 'masterseq');
select * from londiste.root_check_seqs('seqbranch');
select * from londiste.get_seq_list('seqbranch');
select * from londiste.local_remove_seq('seqbranch', 'masterseq');
select * from londiste.local_remove_seq('seqbranch', 'masterseq');

-- seq auto-removal
create table seqtable (
    id1 serial primary key,
    id2 bigserial not null
);

select * from londiste.local_add_table('seqroot', 'seqtable');
select * from londiste.local_add_seq('seqroot', 'seqtable_id1_seq');
select * from londiste.local_add_seq('seqroot', 'seqtable_id2_seq');

select * from londiste.get_table_list('seqroot');
select * from londiste.get_seq_list('seqroot');

select * from londiste.local_remove_table('seqroot', 'seqtable');

select * from londiste.get_seq_list('seqroot');
