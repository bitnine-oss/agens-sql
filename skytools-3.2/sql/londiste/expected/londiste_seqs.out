set client_min_messages = 'warning';
\set VERBOSITY 'terse'
--
-- sequences
--
create sequence masterseq;
create sequence slaveseq;
select * from pgq_node.register_location('seqroot', 'rnode', 'dbname=db', false);
 ret_code |      ret_note       
----------+---------------------
      200 | Location registered
(1 row)

select * from pgq_node.create_node('seqroot', 'root', 'rnode', 'londiste_root', null::text, null::int8, null::text);
 ret_code |                           ret_note                            
----------+---------------------------------------------------------------
      200 | Node "rnode" initialized for queue "seqroot" with type "root"
(1 row)

select * from londiste.local_add_seq('seqroot', 'masterseq');
 ret_code |             ret_note             
----------+----------------------------------
      200 | Sequence added: public.masterseq
(1 row)

select * from londiste.local_add_seq('seqroot', 'masterseq');
 ret_code |                 ret_note                 
----------+------------------------------------------
      201 | Sequence already added: public.masterseq
(1 row)

select * from londiste.root_check_seqs('seqroot');
 ret_code |     ret_note      
----------+-------------------
      100 | Sequences updated
(1 row)

select * from londiste.local_remove_seq('seqroot', 'masterseq');
 ret_code |              ret_note              
----------+------------------------------------
      200 | Sequence removed: public.masterseq
(1 row)

select * from londiste.local_remove_seq('seqroot', 'masterseq');
 ret_code |               ret_note               
----------+--------------------------------------
      400 | Sequence not found: public.masterseq
(1 row)

select * from londiste.get_seq_list('seqroot');
 seq_name | last_value | local 
----------+------------+-------
(0 rows)

select ev_id, ev_type, ev_data, ev_extra1 from pgq.event_template where ev_type like '%seq%';
 ev_id |       ev_type       |     ev_data      |    ev_extra1     
-------+---------------------+------------------+------------------
     1 | londiste.update-seq | 30001            | public.masterseq
     2 | londiste.remove-seq | public.masterseq | 
(2 rows)

-- subscriber
select * from pgq_node.register_location('seqbranch', 'subnode', 'dbname=db', false);
 ret_code |      ret_note       
----------+---------------------
      200 | Location registered
(1 row)

select * from pgq_node.register_location('seqbranch', 'rootnode', 'dbname=db', false);
 ret_code |      ret_note       
----------+---------------------
      200 | Location registered
(1 row)

select * from pgq_node.create_node('seqbranch', 'branch', 'subnode', 'londiste_branch', 'rootnode', 1, null::text);
 ret_code |                              ret_note                               
----------+---------------------------------------------------------------------
      200 | Node "subnode" initialized for queue "seqbranch" with type "branch"
(1 row)

select * from londiste.local_add_seq('seqbranch', 'masterseq');
 ret_code |              ret_note              
----------+------------------------------------
      404 | Unknown sequence: public.masterseq
(1 row)

select * from londiste.global_update_seq('seqbranch', 'masterseq', 5);
 ret_code |     ret_note     
----------+------------------
      200 | Sequence updated
(1 row)

select * from londiste.local_add_seq('seqbranch', 'masterseq');
 ret_code |             ret_note             
----------+----------------------------------
      200 | Sequence added: public.masterseq
(1 row)

select * from londiste.root_check_seqs('seqbranch');
 ret_code |    ret_note     
----------+-----------------
      402 | Not a root node
(1 row)

select * from londiste.get_seq_list('seqbranch');
     seq_name     | last_value | local 
------------------+------------+-------
 public.masterseq |          5 | t
(1 row)

select * from londiste.local_remove_seq('seqbranch', 'masterseq');
 ret_code |              ret_note              
----------+------------------------------------
      200 | Sequence removed: public.masterseq
(1 row)

select * from londiste.local_remove_seq('seqbranch', 'masterseq');
 ret_code |               ret_note               
----------+--------------------------------------
      404 | Sequence not found: public.masterseq
(1 row)

-- seq auto-removal
create table seqtable (
    id1 serial primary key,
    id2 bigserial not null
);
select * from londiste.local_add_table('seqroot', 'seqtable');
 ret_code |           ret_note           
----------+------------------------------
      200 | Table added: public.seqtable
(1 row)

select * from londiste.local_add_seq('seqroot', 'seqtable_id1_seq');
 ret_code |                ret_note                 
----------+-----------------------------------------
      200 | Sequence added: public.seqtable_id1_seq
(1 row)

select * from londiste.local_add_seq('seqroot', 'seqtable_id2_seq');
 ret_code |                ret_note                 
----------+-----------------------------------------
      200 | Sequence added: public.seqtable_id2_seq
(1 row)

select * from londiste.get_table_list('seqroot');
   table_name    | local | merge_state | custom_snapshot | table_attrs | dropped_ddl | copy_role | copy_pos | dest_table 
-----------------+-------+-------------+-----------------+-------------+-------------+-----------+----------+------------
 public.seqtable | t     | ok          |                 |             |             |           |        0 | 
(1 row)

select * from londiste.get_seq_list('seqroot');
        seq_name         | last_value | local 
-------------------------+------------+-------
 public.seqtable_id1_seq |      30001 | t
 public.seqtable_id2_seq |      30001 | t
(2 rows)

select * from londiste.local_remove_table('seqroot', 'seqtable');
 ret_code |            ret_note            
----------+--------------------------------
      200 | Table removed: public.seqtable
(1 row)

select * from londiste.get_seq_list('seqroot');
 seq_name | last_value | local 
----------+------------+-------
(0 rows)

